client/msgraph.go

```diff
@@ -41,63 +96,138 @@ func NewMSGraphApplicationClient(graphURI string, creds azcore.TokenCredential)
 	if err != nil {
 		return nil, err
 	}
-
-	adapter, err := msgraphsdkgo.NewGraphRequestAdapter(authProvider)
-	if err != nil {
-		return nil, err
-	}
-
-	adapter.SetBaseUrl(fmt.Sprintf("%s/v1.0", graphURI))
-	client := msgraphsdkgo.NewGraphServiceClient(adapter)
-
 	ac := &AppClient{
-		client: client,
+		baseUrl:      fmt.Sprintf("%s/v1.0", graphURI),
+		authProvider: authProvider,
 	}
+
 	return ac, nil
 }
 
-func (c *AppClient) GetApplication(ctx context.Context, clientID string) (models.Applicationable, error) {
+func (c *AppClient) GetApplication(ctx context.Context, clientID string) (MSGraphApplication, error) {
 	filter := fmt.Sprintf("appId eq '%s'", clientID)
-	req := applications.ApplicationsRequestBuilderGetRequestConfiguration{
-		QueryParameters: &applications.ApplicationsRequestBuilderGetQueryParameters{
-			Filter: &filter,
-		},
+	query := "$filter=" + url.QueryEscape(filter)
+	u, err := url.Parse(fmt.Sprintf("%s/applications?%s", c.baseUrl, query))
+	if err != nil {
+		return nil, err
 	}
-
-	resp, err := c.client.Applications().Get(ctx, &req)
+	req := abs.NewRequestInformation()
+	req.SetUri(*u)
+	err = c.authProvider.AuthenticateRequest(ctx, req, nil)
 	if err != nil {
 		return nil, err
 	}
+	headers := map[string][]string{
+		"accept":        {"application/json"},
+		"authorization": req.Headers.Get("Authorization"),
+	}
+	request := &http.Request{
+		Method: "GET",
+		URL:    u,
+		Header: headers,
+	}
 
-	apps := resp.GetValue()
-	if len(apps) == 0 {
+	resp, err := http.DefaultClient.Do(request)
```

Comment:
Should we use https://github.com/hashicorp/go-cleanhttp?

Reply:
Yep! I rushed this out, but if we intended to go this route, I'd refactor this to use a common http client and DRY a few things.

But, it sounds like we're looking to use the msgraph SDK in more places, so merging this wouldn't really change anything. I'll close for now but keep the branch.